File: src/main/java/net/spartanb312/cursa/event/events/client/GuiScreenEvent.java
package net.spartanb312.cursa.event.events.client;

import net.spartanb312.cursa.event.CursaEvent;
import net.minecraft.client.gui.GuiScreen;

public class GuiScreenEvent extends CursaEvent {

    private final GuiScreen screen;

    public GuiScreenEvent(GuiScreen screen) {
        this.screen = screen;
    }

    public GuiScreen getScreen() {
        return screen;
    }

    public static class Displayed extends GuiScreenEvent {
        public Displayed(GuiScreen screen) {
            super(screen);
        }
    }

    public static class Closed extends GuiScreenEvent {
        public Closed(GuiScreen screen) {
            super(screen);
        }
    }

}

File: src/main/java/club/deneb/client/event/events/client/GuiScreenEvent.kt
package club.deneb.client.event.events.client

import net.minecraft.client.gui.GuiScreen
import club.deneb.client.event.MinecraftEvent

open class GuiScreenEvent(var screen: GuiScreen) : MinecraftEvent() {
    class Displayed(screen: GuiScreen) : GuiScreenEvent(screen)
    class Closed(screen: GuiScreen) : GuiScreenEvent(screen)
}